src/test/java/uk/gov/ida/eventemitter/EventEmitterIntegrationTest.java

```diff
@@ -1,73 +1,202 @@
-                    CONFIGURATION_ENABLED,
-                    ACCESS_KEY_ID,
-                    ACCESS_SECRET_KEY,
-                    Regions.EU_WEST_2,
-                    QUEUE_ACCOUNT_ID,
-                    SOURCE_QUEUE_NAME,
-                    KEY);
-            }
-        }, Modules.override(new EventEmitterModule()).with(new TestEventEmitterModule()));
 
-        sqs = AmazonHelper.getInstanceOfAmazonSqs(injector);
-        AmazonHelper.createSourceQueue(sqs, SOURCE_QUEUE_NAME);
-        queueUrl = AmazonHelper.getQueueUrl(injector);
+        injector = EventEmitterTestHelper.createTestConfiguration(CONFIGURATION_ENABLED,
+                ACCESS_KEY_ID,
+                ACCESS_SECRET_KEY,
+                Regions.EU_WEST_2,
+                URI.create(apiGatewayStub.baseUri().build().toString() + AUDIT_EVENTS_API_RESOURCE));
+
     }
 
-    @AfterClass
-    public static void tearDown() {
-        sqs.deleteQueue(queueUrl);
+    @Before
+    public void setUpLogAppender() {
+        testAppender.start();
+        Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        logger.addAppender(testAppender);
+        apiGatewayStub.reset();
+    }
+
+    @After
+    public void tearDownLogAppender() throws Exception {
+        testAppender.stop();
+        TestAppender.events.clear();
+        Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        logger.detachAppender(testAppender);
+        apiGatewayStub.reset();
     }
 
     @Test
-    public void shouldEncryptMessageUsingEventEncrypterAndSendToSQS() throws Exception {
+    public void shouldEncryptMessageUsingEventEncrypterAndSendToAPIGateway() throws Exception {
+
+
         final EventEmitter eventEmitter = injector.getInstance(EventEmitter.class);
         final Event event = aTestEventMessage().build();
 
+        apiGatewayStub.register(AUDIT_EVENTS_API_RESOURCE, HttpResponse.HTTP_200.getStatusCode());
+
         eventEmitter.record(event);
 
-        final Message message = AmazonHelper.getAMessageFromSqs(sqs, queueUrl);
-        sqs.deleteMessage(queueUrl, message.getReceiptHandle());
+        final String message = new String(apiGatewayStub.getLastRequest().getEntityBytes());
         final TestDecrypter<TestEvent> decrypter = new TestDecrypter(KEY, injector.getInstance(ObjectMapper.class));
-        final Event actualEvent = decrypter.decrypt(message.getBody(), TestEvent.class);
+        final Event actualEvent = decrypter.decrypt(message, TestEvent.class);
 
         assertThat(actualEvent).isEqualTo(event);
+        assertThat(TestAppender.events.get(0).toString()).contains(
+                String.format("Sent Event Message [Event Id: %s]",
+                        event.getEventId()));
+    }
+
+    @Test
+    public void shouldFailSilentlyWithIncorrectResource() throws Exception {
+
+        final EventEmitter eventEmitter = injector.getInstance(EventEmitter.class);
```

Comment:
Duplicate code.

Reply:
The statement `final EventEmitter eventEmitter = injector.getInstance(EventEmitter.class);` appears in more than one methods in this class.  We can make `private static final EventEmitter eventEmitter = injector.getInstance(EventEmitter.class);` as part of the `EventEmitterIntegrationTest` class and update the methods to use the variable.